src/PgNamed.hs

```diff
@@ -189,6 +191,33 @@ queryNamed conn qNamed params =
     withNamedArgs qNamed params >>= \(q, actions) ->
         liftIO $ PG.query conn q (toList actions)
 
+{- | Queries the database with a given row parser, 'PG.Query', and named parameters
+and expects a list of rows in return.
+
+Sometimes, instead of using Haskell's derived`FromRow` instance to parse tuples returned
+by @postgreSQL@ , we may wish to define our own row parser. In this case, the
```

Comment:
I'm afraid it's not clear enough why users might want to provide their own `RowParser` instead of using `FromRow` instance

Reply:
How about something like this:

queryNamedWith allows the user to being a custom RowParser for scenarios where the columns that are being parsed can be interpreted in more than one way but still map back to the same data structure.

One example of this if parsing some columns differently based on the values of previous columns, like using a different JSON parser for a json column based on some categorical column's value parsed earlier.

Reply:
I would write the following motivation:

1. There is already `FromRow` instance for the type you want, so instead of creating `newtype` you just create custom `RowParser` for convenience. Basically, when you want multiple `FrowRow` instances for the same type and don't want to deal with `newtype` wrapping and wrapping. Though, introducing `newtype` would be a better idea.
2. You have a lot of custom `RowParser`'s with non-trivial logic and you want to share code between them to avoid code duplication, but introducing data types is less convenient.

Reply:
Thanks for your inputs @arbus  and @chshersh . I've rewritten the documentation which combines what both of you guys said, and added a simple example as well. Let me know what you think!

Reply:
@arbus and @chshersh is this good to go? If not, let me know, and I'll make the required changes